### DISCUSSION TITLE
Expand useMediaQuery API

### DISCUSSION THREAD

**Discussion by disillusionedowl** [mantinedev/mantine#4010] [Feature requests]
Calling `useMediaQuery`, according to docs, now works like this:

```js
useMediaQuery('(min-width: 60em)');`
```

Downsides are it expects you to understand and write the (partial) media query syntax and also hard-code the size value(s). I quickly put together a wrapper, nothing special, but could be a source of basic inspiration for future development, that is more straightforward and also connects to the dynamically defined breakpoints from `theme`. Maybe someone here will find it useful:

Usage:

```js
useMediaQuery( [ [ 'min', 'sm' ], [ 'max', 'lg' ] ] );
```

Source:

```js
import { useMantineTheme } from '@mantine/core';
import { useMediaQuery as useMantineMediaQuery } from '@mantine/hooks';

export function useMediaQuery( rules ) {
	const theme = useMantineTheme();
	const query = rules.map( rule => {
		const compareFunction = ( () => {
			if ( rule[0] === 'min' ) {
				return theme.fn.largerThan;
			} else if ( rule[0] === 'max' ) {
				return theme.fn.smallerThan;
			}
		} )();
		return compareFunction( rule[1] ).replace( '@media ', '' );
	} ).join( ' and ' );
	return useMantineMediaQuery( query );
}
```

**Comment by rtivital** [comment]
`@mantine/hooks` package does not depend on `@mantine/core`, and thus it cannot get access to theme context. Furthermore, theme integration for this hook is not planned because it should not be used in most cases as it is not fully compatible with SSR.

**Reply by disillusionedowl** [reply]
Thanks for the details @rtivital . Docs already mention the SSR downside which is good:

_The approach with [MediaQuery](https://mantine.dev/core/media-query/) component will work with SSR. If you do not need SSR support the more cleaner way to change size is to use [use-media-query](https://mantine.dev/hooks/use-media-query/) hook_

Or maybe leaving this hook as-is and adding a different hook into `@mantine/core` with some logic duplication. Could be called e.g. `useBreakpoint()`. But I am fine with the above wrapper for my needs (no SSR), just thought I would mention it and share my current approach here. Big thanks for this wonderful library!
